### DISCUSSION TITLE
ScriptReference/UIElements.VisualElement-pickingMode

### DISCUSSION THREAD

**Discussion by giscus** [Developer-Notes-Extension/Unity-Notes#675] [Notes]
# ScriptReference/UIElements.VisualElement-pickingMode

https://docs.unity3d.com/ScriptReference/UIElements.VisualElement-pickingMode.html

**Comment by vertxxyz** [comment]
Note that if you're cloning UXML you can't set pickingMode in a constructor, because the Factory's UxmlTraits will come along and overwrite it.
See [my forum thread](https://forum.unity.com/threads/pickingmode-is-overridden-when-using-elements-in-uxml.1611720/).

To fix this issue you can create your own `UxmlTraits`, eg.
```csharp
public class BetterUxmlTraits : VisualElement.UxmlTraits
{
	public override void Init(VisualElement ve, IUxmlAttributes bag, CreationContext cc)
	{
		PickingMode prevPickingMode = ve.pickingMode;
		base.Init(ve, bag, cc);
		// Respect the pickingMode set in the constructor.
		if (!bag.TryGetAttributeValue(m_PickingMode.name, out _))
			ve.pickingMode = prevPickingMode;
	}
}
```
which you can subclass and provide to your factory arguments.

Sadly in 2023/6+, due to all this being code generated from attributes I'd probably recommend a different workaround.
